src/Libraries/Microsoft.Extensions.AI.Abstractions/ChatCompletion/ChatResponseExtensions.cs

```diff
@@ -0,0 +1,304 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+
+#pragma warning disable S109 // Magic numbers should not be used
+#pragma warning disable S1121 // Assignments should not be made from within sub-expressions
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Provides extension methods for working with <see cref="ChatResponse"/> and <see cref="ChatResponseUpdate"/> instances.
+/// </summary>
+public static class ChatResponseExtensions
+{
+    /// <summary>Adds all of the messages from <paramref name="response"/> into <paramref name="list"/>.</summary>
+    /// <param name="list">The destination list into which the messages should be added.</param>
+    /// <param name="response">The response containing the messages to add.</param>
+    /// <exception cref="ArgumentNullException"><paramref name="list"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentNullException"><paramref name="response"/> is <see langword="null"/>.</exception>
+    public static void AddMessages(this IList<ChatMessage> list, ChatResponse response)
```

Comment:
From a discoverability point of view, I suspect that:

```cs
response.AppendTo(history);
```

... would be more obvious than:

```cs
history.AddMessages(response);
```

... on the grounds that `IList<T>` wouldn't be expected to know anything about `ChatResponse`, whereas a `ChatResponse`  would be expected to know about an `IList<ChatMessage>`.

I know that for people copy-pasting samples it won't make a difference either way. But for people coming at it from first principles it may be clearer.

This isn't a big deal. Will leave up to you whether you swap it round or not.

Reply:
Actually having seen more of the usage samples below now, I'm really not bothered about this. It feels fine either way.